lib/widgets/message_list.dart

```diff
@@ -781,58 +783,81 @@ class MessageWithPossibleSender extends StatelessWidget {
     final time = _kMessageTimestampFormat
       .format(DateTime.fromMillisecondsSinceEpoch(1000 * message.timestamp));
 
+    final Widget senderWidget;
+    if (item.showSender) {
+      senderWidget = Padding(
+        padding: const EdgeInsets.only(top: 4, left: 16, right: 16),
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          crossAxisAlignment: CrossAxisAlignment.baseline,
+          textBaseline: TextBaseline.alphabetic,
+          children: [
+            Flexible(
+              child: GestureDetector(
+                onTap: () => Navigator.push(context,
+                  ProfilePage.buildRoute(context: context,
+                    userId: message.senderId)),
+                child: Padding(
+                  padding: const EdgeInsets.symmetric(vertical: 2),
+                  child: Row(
+                    children: [
+                      Avatar(size: 32, borderRadius: 3, userId: message.senderId),
+                      const SizedBox(width: 8),
+                      Flexible(
+                        child: Text(message.senderFullName, // TODO get from user data
+                          style: const TextStyle(
+                            fontFamily: 'Source Sans 3',
+                            fontSize: 18,
+                            height: (22 / 18),
+                          ).merge(weightVariableTextStyle(context, wght: 600,
+                              wghtIfPlatformRequestsBold: 900)),
+                          overflow: TextOverflow.ellipsis),
+                      ),
+                    ]),
+                )),
+            ),
+            Text(time,
```

Comment:
The only place we use the `time` local is now inside this `item.showSender` conditional. So that local's definition can be moved inside the conditional too.